### DISCUSSION TITLE
Import Existing Wireguard Configurations (without Client PrivateKey)

### DISCUSSION THREAD

**Discussion by willtho89** [wg-easy/wg-easy#791] [Ideas]
Hi,

I'm looking at replacing my existing wireguard setup with this. I have quite a few devices already and my configuration on the server does not include the clients private keys (or any preshared keys (see https://github.com/wg-easy/wg-easy/pull/763).

I'm wondering if you would be open to changing the code base to be able to have client configs without privateKey and preSharedKey. e.g. 
```json
{
  "server": {
    "privateKey": "XXX",
    "publicKey": "YYY",
    "address": "10.8.0.1"
  },
  "clients": {
    "825486b1-f21f-401f-b3bb-f665e240edb5": {
      "name": "Test",
      "address": "10.8.0.5",
      "publicKey": "zSE446rzWn9swu5Cj7hERV86w745fZsbtcR4XCtiVAM=",
      "createdAt": "2024-01-26T10:02:15.211Z",
      "updatedAt": "2024-01-26T10:02:15.211Z",
      "enabled": true
    }
  }
}
```
This would of course mean, that an admin could not export the client config for these clients. I would propose to disable the download/qr button for these configurations.

If you are open to it, I would create a PR for this scenario.

Best
Thomas

**Comment by pheiduck** [comment]
> If you are open to it, I would create a PR for this scenario.

Sure, feel free to open a PR, but it should not break any other functionality

**Comment by willtho89** [comment]
PR created in #793
